specs/sequencer-app.md

```diff
@@ -41,143 +42,306 @@ idea of "messages", where every transaction can contain multiple messages. The
 benefit of this is that it guarantees that multiple actions can be executed
 atomically.
 
-The structure of an unsigned transaction is as follows:
+The structure of a transaction body is as follows:
 
 ```rust
-pub struct Unsigned {
-    pub(crate) nonce: Nonce,
-    pub(crate) actions: Vec<Action>,
+pub struct TransactionBody {
+    params: TransactionParams,
+    actions: Actions,
 }
 ```
 
-`Nonce` is an incrementing value that represents how many transactions have been
-previously submitted by this account; thus it starts at 0, and must strictly
-increase by 1 for each transaction.
-
-The structure of a signed transaction is as follows:
+Transaction parameters and actions are defined as the following:
 
 ```rust
-pub struct Signed {
-    /// transaction signature
-    pub(crate) signature: ed25519_consensus::Signature,
-    /// the public key of the signer
-    pub(crate) public_key: ed25519_consensus::VerificationKey,
-    /// the transaction that was signed
-    pub(crate) transaction: Unsigned,
+pub struct TransactionParams {
+    nonce: u32,
+    chain_id: String,
 }
-```
 
-The address corresponding to the signer is derived from the
-`ed25519_consensus::VerificationKey` (i.e. the public key).
-
-### Actions
-
-There are currently 2 types of actions implemented.
-
-The first, part of the `accounts` component, is a value-transfer action:
-
-```rust
-pub struct Transfer {
-    to: Address,
-    amount: Balance,
+pub(crate) struct Actions {
+    group: Group, // actions are grouped by "bundleability" and authority level
+    inner: Vec<Action>,
 }
 ```
 
-This action transfers the specified amount funds from the sender to the
-recipient.
+`Nonce` is an incrementing value that represents how many transactions have been
+previously submitted by the currently submitting account; thus it starts at 0,
+and must strictly increase by 1 for each following transaction.
 
-The second, part of the `sequence` component, accepts generic data with a
-specified chain ID:
+The structure of a (signed) transaction is as follows:
 
 ```rust
-pub struct Action {
-    pub(crate) chain_id: Vec<u8>,
-    pub(crate) data: Vec<u8>,
+pub struct Transaction {
+    /// transaction signature
+    signature: ed25519_consensus::Signature,
+    /// the verification key of the signer
+    verification_key: ed25519_consensus::VerificationKey,
+    /// the transaction body which is signed
+    body: TransactionBody,
+    /// a bytes representation of the transaction body
+    body_bytes: Bytes,
 }
 ```
 
-Note that this action does not have any effect on the state of the sequencer; it
-is simply ordered by the sequencer and placed into a block.
+The address corresponding to the signer is derived from the
+`ed25519_consensus::VerificationKey` (i.e. the public key).
 
-## ABCI block lifecycle
+## Actions
```

Comment:
should we link to the [buf page](https://buf.build/astria/protocol-apis/docs/main:astria.protocol.transaction.v1) here? the actions are documented there as well,